[DOC] Restructure Pyroscope documentation and share content (#3798)
grafana/pyroscope: docs/sources/configure-server/about-configurations.md

If you are running Pyroscope on Kubernetes, you can achieve this by storing the configuration file in a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) and mounting it in each Pyroscope container.

This helps avoid a common misconfiguration pitfall: while certain configuration parameters might look like they’re only needed by one type of component, they might in fact be used by multiple components.
For example, the `-distributor.replication-factor` CLI flag is not only required by ingesters, but also by distributors and queriers.

By using a single configuration file, you ensure that each component gets all the configuration that it needs without needing to track which parameter belongs to which component.
There is no harm in passing a configuration that is specific to one component (such as an ingester) to another component (such as a querier).
In such a case, the configuration is simply ignored.

If necessary, you can use advanced CLI flags to override specific values on a particular Pyroscope component or replica.

> This can be helpful if you want to change a parameter that's specific to a certain component, without having to do a full restart of all other components.

Reviewer: ```suggestion
This is helpful if you want to change a parameter that's specific to a certain component, without having to do a full restart of all other components.
```
